nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/ListAzureBlobStorage.java

```diff
@@ -0,0 +1,180 @@
+        @WritesAttribute(attribute = "azure.timestamp", description = "The timestamp in Azure for the blob"),
+        @WritesAttribute(attribute = "mime.type", description = "MimeType of the content"),
+        @WritesAttribute(attribute = "lang", description = "Language code for the content"),
+        @WritesAttribute(attribute = "azure.blobtype", description = "This is the type of blob and can be either page or block type") })
+@Stateful(scopes = { Scope.LOCAL, Scope.CLUSTER }, description = "After performing a listing of blobs, the timestamp of the newest blob is stored. "
+        + "This allows the Processor to list only blobs that have been added or modified after " + "this date the next time that the Processor is run.")
+public class ListAzureBlobStorage extends AbstractListProcessor<BlobInfo> {
+
+    private static final PropertyDescriptor PREFIX = new PropertyDescriptor.Builder().name("Prefix").description("Search prefix for listing").addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(true).required(false).build();
+
+    private static final List<PropertyDescriptor> PROPERTIES = Collections.unmodifiableList(Arrays.asList(AzureConstants.ACCOUNT_NAME, AzureConstants.ACCOUNT_KEY, AzureConstants.CONTAINER, PREFIX));
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTIES;
+    }
+
+    @Override
+    protected Map<String, String> createAttributes(BlobInfo entity, ProcessContext context) {
+        final Map<String, String> attributes = new HashMap<>();
+        attributes.put("azure.etag", entity.getEtag());
+        attributes.put("azure.primaryUri", entity.getPrimaryUri());
+        attributes.put("azure.secondaryUri", entity.getSecondaryUri());
+        attributes.put("azure.blobname", entity.getName());
+        attributes.put("azure.blobtype", entity.getBlobType());
+        attributes.put("azure.length", String.valueOf(entity.getLength()));
+        attributes.put("azure.timestamp", String.valueOf(entity.getTimestamp()));
+        attributes.put("mime.type", entity.getContentType());
+        attributes.put("lang", entity.getContentLanguage());
+
+        return attributes;
+    }
+
+    @Override
+    protected String getPath(final ProcessContext context) {
+        return context.getProperty(AzureConstants.CONTAINER).evaluateAttributeExpressions().getValue();
+    }
+
+    @Override
+    protected boolean isListingResetNecessary(final PropertyDescriptor property) {
+        // re-list if configuration changed, but not when security keys are rolled (not included in the condition)
+        return PREFIX.equals(property)
+                   || AzureConstants.ACCOUNT_NAME.equals(property)
+                   || AzureConstants.CONTAINER.equals(property);
+    }
+
+    @Override
+    protected Scope getStateScope(final ProcessContext context) {
+        return Scope.CLUSTER;
```

Comment:
I talked to Mark Payne and since the cluster scope is explicitly used, he recommended that the local scope be removed from the annotiation, so I'll make that update.